plugin/src/main/java/io/jenkins/plugins/casc/yaml/DefaultMergeStrategy.java

```diff
@@ -0,0 +1,66 @@
+package io.jenkins.plugins.casc.yaml;
+
+import hudson.Extension;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import io.jenkins.plugins.casc.snakeyaml.nodes.MappingNode;
+import io.jenkins.plugins.casc.snakeyaml.nodes.Node;
+import io.jenkins.plugins.casc.snakeyaml.nodes.NodeId;
+import io.jenkins.plugins.casc.snakeyaml.nodes.NodeTuple;
+import io.jenkins.plugins.casc.snakeyaml.nodes.ScalarNode;
+import io.jenkins.plugins.casc.snakeyaml.nodes.SequenceNode;
+import java.util.Iterator;
+
+@Extension
+public class DefaultMergeStrategy implements MergeStrategy {
```

Comment:
I think `Default` is a horrible name it does not describe the merge behavior.

Reply:
StrategicMerge? It’s the name kustomize used for ‘smart’ merge

Reply:
It just throws an exception when there's conflict exists.
What about `IrreconcilableMergeStrategy`?